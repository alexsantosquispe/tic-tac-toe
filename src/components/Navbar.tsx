export const Navbar = () => {
  return (
    <nav className="flex h-14 w-full items-center justify-between border-b border-neutral-200 bg-white px-8 text-lg font-bold text-neutral-600 shadow">
      <p className="text-slate-800">
        TIC<span className="text-rose-600">&nbsp;TAC&nbsp;</span>TOE
      </p>
    </nav>
  );
};
