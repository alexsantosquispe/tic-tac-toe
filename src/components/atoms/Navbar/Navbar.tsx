import { Title } from '../Title';

export const Navbar = () => {
  return (
    <nav className="fixed flex h-14 w-full items-center justify-between border-b border-neutral-200 px-8 font-semibold shadow dark:border-white/20">
      <Title className="text-lg md:text-xl" />
    </nav>
  );
};
