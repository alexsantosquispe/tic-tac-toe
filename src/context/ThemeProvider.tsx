import { useEffect, useState, type ReactNode } from 'react';
import {
  applyTheme,
  getStoredTheme,
  getSystemTheme,
  setStoredTheme
} from '../utils/theme.utils';
import { THEME_TYPES, ThemeContext, type ThemeType } from './ThemeContext';

interface ThemeProviderProps {
  children: ReactNode;
}

const ThemeProvider = ({ children }: ThemeProviderProps) => {
  const initialTheme = getStoredTheme();
  const [currentTheme, setCurrentTheme] = useState<ThemeType>(
    initialTheme ?? THEME_TYPES.SYSTEM
  );

  useEffect(() => {
    const theme =
      currentTheme === THEME_TYPES.SYSTEM ? getSystemTheme() : currentTheme;
    applyTheme(theme);
    setStoredTheme(currentTheme);
  }, [currentTheme]);

  useEffect(() => {
    const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
    const handleChange = (e: MediaQueryListEvent) => {
      if (getStoredTheme() === THEME_TYPES.SYSTEM) {
        applyTheme(e.matches ? THEME_TYPES.DARK : THEME_TYPES.LIGHT);
      }
    };
    mediaQuery.addEventListener('change', handleChange);
    return () => mediaQuery.removeEventListener('change', handleChange);
  }, []);

  return (
    <ThemeContext.Provider
      value={{ theme: currentTheme, setTheme: setCurrentTheme }}
    >
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;
