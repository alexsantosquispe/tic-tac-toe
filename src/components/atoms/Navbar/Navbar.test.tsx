import { render } from '@testing-library/react';
import ThemeProvider from '../../../context/ThemeProvider';
import Navbar from './Navbar';

describe('Navbar', () => {
  it('should render the navbar component correctly', () => {
    const component = render(
      <ThemeProvider>
        <Navbar />
      </ThemeProvider>
    );

    expect(component).toMatchSnapshot();
  });
});
