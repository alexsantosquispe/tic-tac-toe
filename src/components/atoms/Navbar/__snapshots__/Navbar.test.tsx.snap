// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Navbar should render the navbar component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="fixed flex h-14 w-full items-center justify-between border-b border-neutral-200 px-4 font-semibold shadow md:px-8 dark:border-white/20"
      >
        <span
          class="font-medium text-lg md:text-xl"
        >
          TIC-
          <span
            class="text-rose-600 dark:text-rose-500"
          >
            TAC
          </span>
          -TOE
        </span>
        <div
          class="relative flex items-center gap-1 rounded-full border border-neutral-200 p-1 dark:border-white/20"
        >
          <div
            class="absolute top-1 left-1 h-7 w-7 rounded-full bg-rose-600 transform transition-transform duration-200 ease-in-out translate-x-0"
            data-testid="selected-indicator"
          />
          <button
            aria-label="system button"
            class="relative z-10 flex h-7 w-7 items-center justify-center rounded-full hover:cursor-pointer text-white"
          >
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 5a2 2 0 0 1 2 2v8.526a2 2 0 0 0 .212.897l1.068 2.127a1 1 0 0 1-.9 1.45H3.62a1 1 0 0 1-.9-1.45l1.068-2.127A2 2 0 0 0 4 15.526V7a2 2 0 0 1 2-2z"
              />
              <path
                d="M20.054 15.987H3.946"
              />
            </svg>
          </button>
          <button
            aria-label="light button"
            class="relative z-10 flex h-7 w-7 items-center justify-center rounded-full text-neutral-600 hover:cursor-pointer"
          >
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="4"
              />
              <path
                d="M12 2v2"
              />
              <path
                d="M12 20v2"
              />
              <path
                d="m4.93 4.93 1.41 1.41"
              />
              <path
                d="m17.66 17.66 1.41 1.41"
              />
              <path
                d="M2 12h2"
              />
              <path
                d="M20 12h2"
              />
              <path
                d="m6.34 17.66-1.41 1.41"
              />
              <path
                d="m19.07 4.93-1.41 1.41"
              />
            </svg>
          </button>
          <button
            aria-label="dark button"
            class="relative z-10 flex h-7 w-7 items-center justify-center rounded-full text-neutral-600 hover:cursor-pointer"
          >
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.985 12.486a9 9 0 1 1-9.473-9.472c.405-.022.617.46.402.803a6 6 0 0 0 8.268 8.268c.344-.215.825-.004.803.401"
              />
            </svg>
          </button>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="fixed flex h-14 w-full items-center justify-between border-b border-neutral-200 px-4 font-semibold shadow md:px-8 dark:border-white/20"
    >
      <span
        class="font-medium text-lg md:text-xl"
      >
        TIC-
        <span
          class="text-rose-600 dark:text-rose-500"
        >
          TAC
        </span>
        -TOE
      </span>
      <div
        class="relative flex items-center gap-1 rounded-full border border-neutral-200 p-1 dark:border-white/20"
      >
        <div
          class="absolute top-1 left-1 h-7 w-7 rounded-full bg-rose-600 transform transition-transform duration-200 ease-in-out translate-x-0"
          data-testid="selected-indicator"
        />
        <button
          aria-label="system button"
          class="relative z-10 flex h-7 w-7 items-center justify-center rounded-full hover:cursor-pointer text-white"
        >
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 5a2 2 0 0 1 2 2v8.526a2 2 0 0 0 .212.897l1.068 2.127a1 1 0 0 1-.9 1.45H3.62a1 1 0 0 1-.9-1.45l1.068-2.127A2 2 0 0 0 4 15.526V7a2 2 0 0 1 2-2z"
            />
            <path
              d="M20.054 15.987H3.946"
            />
          </svg>
        </button>
        <button
          aria-label="light button"
          class="relative z-10 flex h-7 w-7 items-center justify-center rounded-full text-neutral-600 hover:cursor-pointer"
        >
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <path
              d="M12 2v2"
            />
            <path
              d="M12 20v2"
            />
            <path
              d="m4.93 4.93 1.41 1.41"
            />
            <path
              d="m17.66 17.66 1.41 1.41"
            />
            <path
              d="M2 12h2"
            />
            <path
              d="M20 12h2"
            />
            <path
              d="m6.34 17.66-1.41 1.41"
            />
            <path
              d="m19.07 4.93-1.41 1.41"
            />
          </svg>
        </button>
        <button
          aria-label="dark button"
          class="relative z-10 flex h-7 w-7 items-center justify-center rounded-full text-neutral-600 hover:cursor-pointer"
        >
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.985 12.486a9 9 0 1 1-9.473-9.472c.405-.022.617.46.402.803a6 6 0 0 0 8.268 8.268c.344-.215.825-.004.803.401"
            />
          </svg>
        </button>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
